src/devices/Arduino/ArduinoBoard.cs

```diff
@@ -0,0 +1,413 @@
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.Device.Analog;
+using System.Text;
+using System.Device.Gpio;
+using System.Device.I2c;
+using System.Device.Pwm;
+using System.Device.Spi;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Ports;
+using System.Threading;
+using System.Linq;
+using UnitsNet;
+
+namespace Iot.Device.Arduino
+{
+    /// <summary>
+    /// Implements an interface to an arduino board which is running Firmata.
+    /// See documentation on how to prepare your arduino board to work with this.
+    /// Note that the program will run on the PC, so you cannot disconnect the
+    /// Arduino while this driver is connected.
+    /// </summary>
+    public class ArduinoBoard : IDisposable
+    {
+        private SerialPort? _serialPort;
+        private Stream _dataStream;
+        private FirmataDevice? _firmata;
+        private List<SupportedPinConfiguration> _supportedPinConfigurations;
+
+        // Counts how many spi devices are attached, to make sure we enable/disable the bus only when no devices are attached
+        private int _spiEnabled;
+
+        /// <summary>
+        /// Creates an instance of an Ardino board connection using the given stream (typically from a serial port)
+        /// </summary>
+        /// <param name="serialPortStream">A stream to an Arduino/Firmata device</param>
+        public ArduinoBoard(Stream serialPortStream)
+        {
+            _dataStream = serialPortStream;
+            _spiEnabled = 0;
+            _supportedPinConfigurations = new List<SupportedPinConfiguration>();
+            FirmwareName = string.Empty;
+            FirmataVersion = new Version();
+            FirmwareVersion = new Version();
+        }
+
+        /// <summary>
+        /// Creates an instance of the Arduino board connection connected to a serial port
+        /// </summary>
+        /// <param name="portName">Port name. On Windows, this is usually "COM3" or "COM4" for an Arduino attached via USB.
+        /// On Linux, possible values include "/dev/ttyAMA0", "/dev/serial0", "/dev/ttyUSB1", etc.</param>
+        /// <param name="baudRate">Baudrate to use. It is recommended to use at least 115200 Baud.</param>
+        public ArduinoBoard(string portName, int baudRate)
+        {
+            _serialPort = new SerialPort(portName, baudRate);
+            _serialPort.Open();
+            _dataStream = _serialPort.BaseStream;
+            _supportedPinConfigurations = new List<SupportedPinConfiguration>();
+            FirmwareName = string.Empty;
+            FirmataVersion = new Version();
+            FirmwareVersion = new Version();
+        }
+
+        /// <summary>
+        /// Default pin numbering scheme of this board. Always <see cref="PinNumberingScheme.Logical"/>, because the Arduino has only one pin numbering scheme.
+        /// </summary>
+        public PinNumberingScheme DefaultPinNumberingScheme
```

Comment:
NIT: make this internal or private? I don't really see the use of this when it always return the same and no one really seems to call it.

Reply:
As noted elsewhere already: This will later be defined by the base class, where it is not necessarily constant.